runhouse/resources/hardware/cluster.py

```diff
@@ -1128,6 +1130,33 @@ def _start_or_restart_helper(
         # Update the cluster config on the cluster
         self.save_config_to_cluster()
 
+        # Save a limited version of the local ~/.rh config to the cluster with the user's
+        # if such does not exist on the cluster
+        if rns_client.token:
+            user_config = {
+                "token": rns_client.cluster_token(resource_address=rns_client.username),
+                "username": rns_client.username,
+                "default_folder": rns_client.default_folder,
+            }
+
+            condensed_config_yaml_path = (
```

Comment:
i think we can do something like this in just a couple lines:
```
yaml_path = Path(f"{tempfile.mkdtemp()}/config.yaml")
with open(yaml_path, "w") as config_file:
    yaml.safe_dump(user_config, config_file)
```

my worry is messing with the "~/.rh" folder in any capacity